collections/chunk.ts

```diff
@@ -40,24 +40,17 @@
 export function chunk<T>(array: readonly T[], size: number): T[][] {
   if (size <= 0 || !Number.isInteger(size)) {
     throw new Error(
-      `Expected size to be an integer greater than 0 but found ${size}`,
+      `Expected size to be a positive integer, but found ${size}`,
     );
   }
 
-  if (array.length === 0) {
-    return [];
-  }
-
-  const ret = Array.from<T[]>({ length: Math.ceil(array.length / size) });
-  let readIndex = 0;
-  let writeIndex = 0;
-
-  while (readIndex < array.length) {
-    ret[writeIndex] = array.slice(readIndex, readIndex + size);
+  const result: T[][] = [];
+  let index = 0;
 
-    writeIndex += 1;
-    readIndex += size;
+  while (index < array.length) {
+    result.push(array.slice(index, index + size));
+    index += size;
   }
```

Comment:
This changes how we allocate the result array. Are you sure this doesn't cause perf degradation?

Reply:
The difference is negligible.
```ts
import { chunk } from "jsr:@std/collections@0.224.0/chunk";
import { chunk as chunk2 } from "./collections/chunk.ts";

function createArray(length: number) {
  return Array.from({ length }).fill(0);
}

Deno.bench("chunk()", (b) => {
  const array = createArray(1_000_000);
  b.start();
  chunk(array, 3);
  b.end();
});

Deno.bench("chunk2()", (b) => {
  const array = createArray(1_000_000);
  b.start();
  chunk2(array, 3);
  b.end();
});
```
```
asher@Ashers-MacBook-Air deno_std % deno bench bench.ts
Check file:///Users/asher/GitHub/deno_std/bench.ts
cpu: Apple M2
runtime: deno 1.43.1+4ab68df (aarch64-apple-darwin)

file:///Users/asher/GitHub/deno_std/bench.ts
benchmark      time (avg)        iter/s             (min … max)       p75       p99      p995
--------------------------------------------------------------- -----------------------------
chunk()        20.43 ms/iter          48.9   (11.23 ms … 33.67 ms) 25.44 ms 33.67 ms 33.67 ms
chunk2()       20.19 ms/iter          49.5   (11.58 ms … 32.57 ms) 25.34 ms 32.57 ms 32.57 ms
```